src/components/HTMLEngineProvider/HTMLRenderers/PreRenderer/index.js

```diff
@@ -1,45 +1,48 @@
-import React, {useEffect, useRef} from 'react';
-import {ScrollView} from 'react-native-gesture-handler';
-import {View} from 'react-native';
-import _ from 'underscore';
+import React from 'react';
 import withLocalize from '../../../withLocalize';
 import htmlRendererPropTypes from '../htmlRendererPropTypes';
+import BasePreRenderer from './BasePreRenderer';
 
-const PreRenderer = (props) => {
-    const TDefaultRenderer = props.TDefaultRenderer;
-    const defaultRendererProps = _.omit(props, ['TDefaultRenderer']);
-    const ref = useRef();
+class PreRenderer extends React.Component {
+    constructor(props) {
+        super(props);
 
-    const wheelEvent = (event) => {
-        const node = ref.current.getScrollableNode();
+        this.wheelEvent = this.wheelEvent.bind(this);
+        this.ref = React.createRef(null);
```

Comment:
Use Callbacks instead, if possible.

Reply:
@parasharrajat what do you mean by callback here. Can you explain it ?

Reply:
Sorry tagged the wrong line. I meant to use callback refs. e.g.
```
ref={el => this.ref = el} 
```

Reply:
> Sorry tagged the wrong line. I meant to use callback refs. e.g.
> 
> ```
> ref={el => this.ref = el} 
> ```

@parasharrajat No problem. I don't think it's necessary. But if there is such a requirement in the development rules, I can change it.

Reply:
> Yes. we use callbacks.
`ref={el => this.ref = el}`

@parasharrajat sorry this definition is not working. 
Because it is late in callback and component did mount also get error because ref is null there.

Reply:
Works great for me. Try again.

Reply:
> Works great for me. Try again.

@parasharrajat not work with 
`ref={el => this.ref = el}`
work with
` ref={el => this.ref.current = el}`

for this `ref={el => this.ref = el}`
![image](https://user-images.githubusercontent.com/61627150/166477394-43260d8e-3e17-49ec-9917-f6eb29c3a2d1.png)

Reply:
```js
  constructor(props) {
        super(props);

        this.wheelEvent = this.wheelEvent.bind(this);
    }

    componentDidMount() {
        this.ref.getScrollableNode().addEventListener('wheel', this.wheelEvent, true);
    }

    componentWillUnmount() {
        this.ref.getScrollableNode().removeEventListener('wheel', this.wheelEvent, true);
    }

    wheelEvent(event) {
        const node = this.ref.getScrollableNode();
        const checkOverflow = (node.scrollHeight / node.scrollWidth) !== (node.offsetHeight / node.offsetWidth);

        if ((event.currentTarget === node) && checkOverflow) {
            node.scrollLeft += event.deltaX;
            event.preventDefault();
            event.stopPropagation();
        }
    }

    render() {
        return (
            <BasePreRenderer
                // eslint-disable-next-line react/jsx-props-no-spreading
                {...this.props}
                ref={el => this.ref = el}
            />
        );
    }
```

Does this work for you?

Reply:
> ```js
>   constructor(props) {
>         super(props);
> 
>         this.wheelEvent = this.wheelEvent.bind(this);
>     }
> 
>     componentDidMount() {
>         this.ref.getScrollableNode().addEventListener('wheel', this.wheelEvent, true);
>     }
> 
>     componentWillUnmount() {
>         this.ref.getScrollableNode().removeEventListener('wheel', this.wheelEvent, true);
>     }
> 
>     wheelEvent(event) {
>         const node = this.ref.getScrollableNode();
>         const checkOverflow = (node.scrollHeight / node.scrollWidth) !== (node.offsetHeight / node.offsetWidth);
> 
>         if ((event.currentTarget === node) && checkOverflow) {
>             node.scrollLeft += event.deltaX;
>             event.preventDefault();
>             event.stopPropagation();
>         }
>     }
> 
>     render() {
>         return (
>             <BasePreRenderer
>                 // eslint-disable-next-line react/jsx-props-no-spreading
>                 {...this.props}
>                 ref={el => this.ref = el}
>             />
>         );
>     }
> ```
> 
> Does this work for you?

@parasharrajat This doesn't work anyway because the ref is not defined.

Reply:
Did you try it?

Reply:
@parasharrajat I fixed the issue. Converted successfully